delayed_assert/delayed_assert.py

```diff
@@ -39,18 +39,6 @@ def expect(expr, msg=None):
     global _failed_expectations, _is_first_call
     caller = ''
 
-    '''
-    ensure that the call is coming from 'test*' method
-    '''
-    stack_list = inspect.stack()
-    for stack in stack_list:
-        if stack.function.startswith('test'):
-            caller = stack.function
-            break
-
-    if caller == '':
-        raise Exception("Could not identify test method, make sure the call for 'expect' method is originated with 'test' method")
-
```

Comment:
This block ensures that stack traces are reported correctly when running tests in parallel.

I guess this would be a limitation of the current implementation of the library.

Reply:
If you could find a way to identify the test method, that would solve the issue when running test parallely.

Reply:
But how do we come to the conclusion that the caller method should always start with test* if not we throw the error?? It is like forcing the user whoever using this delayed_assert library to name their methods starts with test*. The expect calls can happen from any arbitrary function that's the user choice. I would say, Instead of throwing the exception which is a breaking change, why don't we set the caller to what it was previously incase you couldn't find any method that starts with test* in the stack. Let me change this MR inline to what I am saying.